testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -754,6 +757,20 @@ public void testShowCreateInformationSchemaTable()
                         ")");
     }
 
+    @Test
+    public void testCreateTableNotAllowedInTransaction()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && !hasBehavior(SUPPORTS_MULTI_STATEMENT_WRITES));
+
+        assertThatThrownBy(() ->
+                transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                        .readUncommitted()
+                        .execute(getSession(), session -> {
+                            computeActual(session, "CREATE TABLE create_table_not_allowed (x int)");
```

Comment:
Can you please add a test for INSERT, CREATE SCHEMA, CREATE VIEW, CREATE MATERIALIZED VIEW, REFRESH MATERIALIZED VIEW, CTAS, (ANALYZE?)?

Reply:
Done